src/nsrt_learning/strips_learning/gen_to_spec_learner.py

```diff
@@ -229,7 +226,8 @@ def _backchain_one_pass(
                 else:
                     nec_pnad_set_changed = True
                     pnad = self._spawn_new_pnad(
-                        param_opt_to_general_pnad[option.parent], segment)
+                        self._create_general_pnad_for_option(option.parent),
```

Comment:
do you want to just move this inside of `_spawn_new_pnad`? or are we ever calling `_spawn_new_pnad` with some pnad other than the most general one?

Reply:
agreed, let's move this into `_spawn_new_pnad` since we are now understanding "spawn" to mean "spawn from the most general"